conn.go

```diff
@@ -471,6 +490,19 @@ func (c *Conn) closeDuringStart() {
 	})
 }
 
+// returns the error indicating why Conn has closed
+func (c *Conn) closedErr() error {
+	// an empty ConnError means the connection was closed by the caller
+	var connErr *ConnError
+	if errors.As(c.doneErr, &connErr) && connErr.RemoteErr == nil && connErr.inner == nil {
```

Comment:
Can we just put a comment in here that this function is only safe to call if c.done is closed? Or just move the select {} on c.done into here? 

Otherwise, I believe we've got a data race checking the value of 'doneErr'.

Reply:
Good point about the comment, I'll add one.

Reply:
Done.